code/game/objects/items/weapons/tools/weldingtool.dm

```diff
@@ -109,11 +109,12 @@
 	if(!status && istype(W, /obj/item/stack/material/rods))
 		var/obj/item/stack/material/rods/R = W
 		R.use(1)
-		add_fingerprint(user)
 		user.drop_from_inventory(src)
 		var/obj/item/flamethrower/F = new(get_turf(src), src)
+		F.add_fingerprint(user)
```

Comment:
It's highly unlikely to be correct to manually add fingerprints like this.

Reply:
now prints are also being added everywhere, for an alternative option it needs to be refactored

Reply:
there is no choice, they are added so everywhere

Reply:
I'm not actually sure this one is replicated in equip code, but it likely should be.

Reply:
looked at process ``put_in_hands``, ``drop_from_inventory``, and ``remove_from_mob`` and didn't found where fingerprints has be added

Reply:
Added an add_fingerprint() call to base item equipped().